app/contracts/queries/projects/project_queries/base_contract.rb

```diff
@@ -41,17 +41,36 @@ def self.model
               presence: true,
               length: { maximum: 255 }
 
-    validate :user_is_current_user_and_logged_in
     validate :name_select_included
     validate :existing_selects
+    validate :user_is_logged_in
+    validate :allowed_to_modify_private_query
+    validate :allowed_to_modify_public_query
 
+    protected
 
-    def user_is_current_user_and_logged_in
-      unless user.logged? && user == model.user
+    def user_is_logged_in
+      if !user.logged?
         errors.add :base, :error_unauthorized
       end
     end
 
+    def allowed_to_modify_private_query
+      return if model.public?
+
+      if model.user != user
+        errors.add :base, :can_only_be_modified_by_owner
+      end
```

Comment:
Wouldn't two guards read cleaner:
```suggestion
      return if model.public?
      return if model.user == user

      errors.add :base, :can_only_be_modified_by_owner
```

Reply:
I basically followed the style that is used in the other methods in the class. They usuall go with "auxillary" clauses as guards and the "main" check as an if/unless clause